src/basic/btrfs-util.c

```diff
@@ -144,10 +143,6 @@ int btrfs_subvol_make(const char *path) {
         if (r < 0)
                 return r;
 
-        fd = open_parent(path, O_RDONLY|O_NOCTTY|O_CLOEXEC|O_DIRECTORY);
-        if (fd < 0)
-                return fd;
-
         strncpy(args.name, subvolume, sizeof(args.name)-1);
```

Comment:
I don't like this I must say... you pass the last component of the specified `path` parameter as subvol name to create, thus you appear to assume that the `fd` parameter matches the parent of `path`. This assumption is pretty weird though...

Either this call should insist on taking only filename names instead of paths (i.e. nothing containing a slash), or it should behave like a true _at() family call... But this weird concept that it it accepts full paths as `path` argument but then throws most of it away is just not right...